src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt

```diff
@@ -220,6 +243,28 @@ fun refundTransaction(
     }
 }
 
+fun handleNpgRefundResponse(
+  transaction: BaseTransactionWithRequestedAuthorization,
+  refundResponse: RefundResponseDto,
+  transactionsEventStoreRepository: TransactionsEventStoreRepository<TransactionRefundedData>,
+  transactionsViewRepository: TransactionsViewRepository
+): Mono<BaseTransaction> {
+  logger.info(
+    "Transaction requestRefund for transaction ${transaction.transactionId} NPG refund status [${refundResponse.operationId!!}]")
+
+  return when (refundResponse.operationId) { // TODO Check this response value, which are possible?
+    "operationId" -> { // TODO this value need to be retrieved from transaction event data
```

Comment:
This field is an NPG identifier (an UUID), why this when here?

Reply:
Hi @pietro-tota , nexi specs are a little bit ambiguous in this response.
We know that refundResponse should contain 2 field:

- operationId
- operationDateTime

Both aren't mandatory. So what i guess here is this business logic:
If I receive refundResponse with operationId that matches the operationId of the authorization i'm asking for refunding, i can guess that refund operation is ok. If your doubt is about constant string "operationId", don't forget that this is a placeholder for a data that we'll retrieve from authorization request data when we'll save it from the patch. This task will be faced in the next sprint, so by now I put here a constant.

Reply:
Comparing the docs with the openapi specifications that we have on the other api, I think those fields are mandatory so we can avoid the check (unfortunately we do not have the openapi specifications for refund api )